test/test_mlmodel.py

```diff
@@ -181,3 +182,90 @@ def test_pipeline():
 
     assert expected_shape == enc_norm_samples.shape
     assert enc_norm_samples.select_dtypes(exclude=[np.number]).empty
+
+
+def test_predictions_pt():
+    data_name = "adult"
+    data_catalog = "adult_catalog.yaml"
+    data = DataCatalog(data_name, data_catalog)
+
+    feature_input_order = [
+        "age",
+        "fnlwgt",
+        "education-num",
+        "capital-gain",
+        "capital-loss",
+        "hours-per-week",
+        "sex_Female",
+        "sex_Male",
+        "workclass_Non-Private",
+        "workclass_Private",
+        "marital-status_Married",
+        "marital-status_Non-Married",
+        "occupation_Managerial-Specialist",
+        "occupation_Other",
+        "relationship_Husband",
+        "relationship_Non-Husband",
+        "race_Non-White",
+        "race_White",
+        "native-country_Non-US",
+        "native-country_US",
+    ]
+    model = MLModelCatalog(data, "ann", feature_input_order, backend="pytorch")
```

Comment:
Personally I'm not a big fan of strings to keep properties, here backend="pytorch". Isn't something like enum a better solution?

Reply:
Would there be any big improvements? All I can think of at the moment is, that a user would have to pass an enum with correct values, similar to the current implementation. Wouldn't it be less intuitive?

Reply:
@JohanvandenHeuvel I think its pretty common to expect string values to user-facing apis.E.g. pandas and seaborn/matplotlib do it too, and i am sure sklearn aswell.

E.g.: 
- `pd.merge(how='inner')` https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.merge.html
- `sns.histplot(data=penguins, ..., multiple="stack")` https://seaborn.pydata.org/generated/seaborn.histplot.html#seaborn.histplot

As Sascha said it would be quite a pain for the user to import the corresponding enum first.

----
**But** in general you are right, non-userfacing apis/functions should generally expect enums instead of strings

Reply:
Yeah, it's one of the things I don't really like in python